core/domain/opportunity_services.py

```diff
@@ -361,6 +361,30 @@ def get_voiceover_opportunities(language_code, cursor):
     return opportunities, cursor, more
 
 
+def get_opportunities_by_ids(ids):
+    """Returns a list of ExplorationOpportunitySummary object corresponding to
+    the given list of ids.
+
+    Args:
+        ids: list(str). A list of the opportunity ids.
+
+    Returns:
+        list(ExplorationOpportunitySummary). A list of
+        ExplorationOpportunitySummary object.
+    """
+    exp_opportunity_summary_models = (
+        opportunity_models.ExplorationOpportunitySummaryModel.get_multi(ids))
+    opportunities = {}
+    for exp_opportunity_summary_model in exp_opportunity_summary_models:
+        exp_opportunity_summary = (
+            get_exploration_opportunity_summary_from_model(
+                exp_opportunity_summary_model))
+        opportunities[exp_opportunity_summary.id] = (
+            exp_opportunity_summary.to_dict())
```

Comment:
Don't use to_dict() in domain layer. Return domain objects always; only convert to dict at controller endpoint.